### PR TITLE
Added ability to start a new group chat!

### PR DESCRIPTION
**Pull request by qcmaude**
This fixes Issue #56

### CODE REVIEW

**Comment by qcmaude** [comment]
It's in `sendMessage` because making a new group chat makes a call to the same endpoint (`send_message.php`) and uses very (very) similar form fields. The only big difference is that there is an added iteration through an array of user ids (and some checks put were in place to make sure that everything else remains the same).

File: src/sendMessage.js [hunk 1]
```diff
@@ -103,52 +104,75 @@ module.exports = function(defaultFuncs, api, ctx) {
...
+      // There are three cases here:
+      // 1. threadID is of type array, where we're starting a new group chat with users
+      //    specified in the array.
+      // 2. User is sending a message to a specific user.
+      // 3. No additional form params and the message goes to an existing group chat.
+      if(threadIDType === "Array") {
+        for (var i  = 0; i < threadID.length; i++) {
+          form['message_batch[0][specific_to_list][' + i + ']'] = "fbid:" + threadID[i];
         }
+        form['message_batch[0][specific_to_list][' + (threadID.length + 1) + ']'] = "fbid:" + ctx.userID;
```
Comment (bsansouci) [inline src/sendMessage.js]: Shouldn't this be `threadID.length` instead of `threadID.length + 1`?

File: src/sendMessage.js [hunk 2]
```diff
@@ -103,52 +104,75 @@ module.exports = function(defaultFuncs, api, ctx) {
...
-      api.getUserInfo(threadID, function(err, res) {
-        if (err) {
-          return callback(err);
+    function sendContent(isSingleUser) {
+      // There are three cases here:
+      // 1. threadID is of type array, where we're starting a new group chat with users
+      //    specified in the array.
+      // 2. User is sending a message to a specific user.
+      // 3. No additional form params and the message goes to an existing group chat.
+      if(threadIDType === "Array") {
```
Comment (bsansouci) [inline src/sendMessage.js]: Could you use `isSingleUser` here?

File: src/sendMessage.js [hunk 3]
```diff
@@ -103,52 +104,75 @@ module.exports = function(defaultFuncs, api, ctx) {
...
+        .then(utils.parseAndCheckLogin)
+        .then(function(resData) {
+          if (!resData) {
+            throw {error: "Send message failed."};
+          }
+          if(resData.error) {
+            throw resData;
+          }
+
+          return callback(null, resData.payload.actions.reduce(function(p, v) {
```
Comment (bsansouci) [inline src/sendMessage.js]: This should only contain one thing right?

Comment (Schmavery) [inline src/sendMessage.js]: I think it's safer to do this, considering actions is an array.

File: src/sendMessage.js (hunk 1 above)
Comment (qcmaude) [inline src/sendMessage.js]: LOLOL yep.

**Comment by Schmavery** [comment]
I think this should be safe to merge.  Thanks for updating docs.

**Pull request merged**